utopia-remix/app/routes/internal.projects.$id.github.branches.$owner.$repo.branch.$branch.tsx

```diff
@@ -0,0 +1,76 @@
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node'
+import { ensure, handle, handleOptions, requireUser } from '../util/api.server'
+import { validateProjectAccess } from '../handlers/validators'
+import type { Params } from '@remix-run/react'
+import type { ExistingAsset } from '../types'
+import { UserProjectPermission } from '../types'
+import { Status } from '../util/statusCodes'
+import { getGithubAuthentication } from '../models/githubAuthentication.server'
+import { getBranchProjectContents } from '../util/github-branch-contents.server'
+import { newOctokitClient, wrapGithubAPIRequest } from '../util/github'
+
+export function loader(args: LoaderFunctionArgs) {
+  return handle(args, {
+    OPTIONS: handleOptions,
+  })
+}
+
+export function action(args: ActionFunctionArgs) {
+  return handle(args, {
+    POST: {
+      validator: validateProjectAccess(UserProjectPermission.CAN_MANAGE_PROJECT, {
+        getProjectId: (params) => params.id,
+      }),
+      handler: handler,
+    },
+  })
+}
+
+export type GetBranchProjectContentsRequest = {
+  existingAssets: ExistingAsset[] | null
+  uploadAssets: boolean
+}
+
+function isBranchContentsRequest(u: unknown): u is GetBranchProjectContentsRequest {
+  const maybe = u as GetBranchProjectContentsRequest
+  return (
+    u != null &&
+    typeof u === 'object' &&
+    maybe.existingAssets != null &&
+    (maybe.uploadAssets == null || Array.isArray(maybe.existingAssets))
```

Comment:
Looks like you've confused `existingAssets` and `uploadAssets` in these two lines. It might just be simpler to add `type: 'GET_BRANCH_PROJECT_CONTENTS_REQUEST` to `GetBranchProjectContentsRequest` to avoid shooting yourself in the foot here

Reply:
dammit nice catch 🤦 
actually yeah adding a union type on a `type` makes a lot more sense, will do

Reply:
https://github.com/concrete-utopia/utopia/pull/5639/commits/f1571b8a3aefc1897e4593aeb2c0d46567fb2ac9